File: hw/jacome/U3/HW22Strategy/JS/controller/SortingContext.js
const BubbleSort = require('../model/BubbleSort');
const InsertionSort = require('../model/InsertionSort');
const QuickSort = require('../model/QuickSort');

class SortingContext {
    constructor() {
        this.ss = null;
    }
    
    sort(data) {
        const size = data.length;
        this.ss = this.setSortStrategy(size);
        return this.ss.sort(data);
    }
    
    setSortStrategy(n) {
        if (n > 0 && n < 30) {
            return new BubbleSort();
        }
        if (n >= 30 && n < 100) {
            return new InsertionSort();
        }
        if (n >= 100) {
            return new QuickSort();
        }
        return new BubbleSort(); // Default case
    }
}

module.exports = SortingContext;

File: hw/paillacho/U3/HW22Strategy/StrategyPython/controller/sorting_context.py
from model.bubble_sort import BubbleSort
from model.insertion_sort import InsertionSort
from model.quick_sort import QuickSort

# Author: Carlos Paillacho


class SortingContext:

    def __init__(self):
        self.ss = None

    def sort(self, data):
        size = len(data)
        self.ss = self.set_sort_strategy(size)
        return self.ss.sort(data)

    def set_sort_strategy(self, n):
        if 0 < n < 30:
            self.ss = BubbleSort()
        elif 30 <= n < 100:
            self.ss = InsertionSort()
        elif n >= 100:
            self.ss = QuickSort()
        return self.ss
